### DISCUSSION TITLE
Avoid changing code until semantic stage has finished

### DISCUSSION THREAD

**Discussion by rakati** [pyccel/pyccel#1217] [General]
It is hard for a user to detect the problem with his code when the semantic errors mention some part of the code or variables' names different than the original code.
**Example**
```python
def func(p:int, normalize:bool=True):
    B = np.ones(p)
    if normalize:
        return p*B
    else:
        return B
```
the error raised by Pyccel is:
```shell
ERROR at annotation (semantic) stage
pyccel:
 |error [semantic]: main.py [60,8]| Arrays which own their data cannot become views on other arrays (Out_0001)
```
this is just a small example, where `Out_0001` is created during the sematic stage, and this may be too complicated for a large code, and the user may struggle a lot because of that.

**Comment by EmilyBourne** [comment]
Is the line number right at least? I think the case of mismatched `return`s should probably have it's own error anyway. It is not at all clear that this error is raised because `return p*B` returns an array and `return B` returns a view.
